### DISCUSSION TITLE
Laravel's `dimensions` validation rule does not work

### DISCUSSION THREAD

**Discussion by gdebrauwer** [livewire/livewire#5229] [7. Bugs (archived)] [answered]
Laravel's `dimensions` validation rule does not work on Livewire's `TemporaryUploadedFile`. This is caused by the fact that the `getRealPath` method returns the file path instead of the file URL.

**Comment by JuanDMeGon** [comment]
Facing the same right now. We would probably need a custom rule implementation for `TemporaryUploadedFile`.

**Comment by joshhanley** [comment]
@gdebrauwer or @JuanDMeGon any chance you could share your code which reproduces the issue with as little code as possible so we can investigate it? Thanks!

**Comment by JuanDMeGon** [comment]
Hey, @joshhanley 

Absolutely! Here are some details:

First, in the `config/livewire.php`, set `temporary_file_upload.disk` as `s3`, so the files are directly uploaded to S3 

Now a tiny component that validates a thumbnail (image)dimensions:

```
<?php

namespace App\Http\Livewire\Media;

use Livewire\Component;
use Livewire\WithFileUploads;

class Create extends Component
{
    use WithFileUploads;

    public $thumbnail;

    protected function rules()
    {
        return [
            'thumbnail' => [
                'image',
                'max:3000',
                'dimensions:max_width=1000,max_height=1000',
            ],
        ];
    }
    public function store()
    {
        $this->validate();

        //Something here
    }

    public function render()
    {
        return view('livewire.test');
    }
}
```

Notice the `dimensions` call in the rules. Because of the direct upload to S3, the dimensions rule is constantly failing, and you won't be able to validate that file dimensions or upload it so.

Checking the `validateDimensions` at the `Illuminate\Validation\Concerns\ValidatesAttributes` it calls `@getimagesize`, which is always returning false, causing the validation to fail.

**Reply by joshhanley** [reply]
@JuanDMeGon great, thanks for sharing! I will see if I can have a play with it when I get the chance 🙂

**Reply by JuanDMeGon** [reply]
Awesome! Let me know if I can help somehow.

From what I've seen it's a tricky one due the nature of the direct upload and the fake file used for it.

Thanks and best wishes. 🙃

**Comment by gdebrauwer** [comment, marked as answer]
Fixed in v2.12.5 en v3
